File: Data Structures & Algorithms/climbing-stairs/submission-0.cpp
class Solution {
public:
    int climbStairs(int n) {
        if(n<=2)return n;
        vector<int> x(n+1);
        x[1]=1;
        x[2]=2;
        for(int i=3;i<=n;i++){
            x[i]=x[i-1]+x[i-2];
        } 
        return x[n];
    }
};

File: Data Structures & Algorithms/hand-of-straights/submission-0.py
class Solution:
    def isNStraightHand(self, hand: List[int], groupSize: int) -> bool:
        if len(hand) % groupSize != 0:
            return False

        count = Counter(hand)

        for start in sorted(count):
            while count[start]>0:
                for num in range(start,start+groupSize):
                    if count[num] == 0:
                        return False
                    
                    count[num] -= 1
                
        return True